cds/modules/deposit/indexer.py

```diff
@@ -43,3 +47,52 @@ def cdsdeposit_indexer_receiver(
     if record['$schema'] in [project_schema, video_schema]:
         json['_cds']['state'] = deposit['_cds']['state']
         json['_files'] = deposit['_files']
+
+
+class CDSRecordIndexer(RecordIndexer):
+
+    def index(self, deposit, action='commit'):
+        project_schema = current_jsonschemas.path_to_url(Project._schema)
+        if action == 'publish':
+            if deposit['$schema'] == project_schema:
```

Comment:
if this method stays so long, maybe extract to a method?
`_index_project()`
`_index_videos(ids)`

Reply:
This could be good. I will check how I can split them.